domain/credential/state/state.go

```diff
@@ -77,8 +87,14 @@ func (st *State) UpsertCloudCredential(ctx context.Context, id credential.ID, cr
 	q := `
 SELECT  cloud_credential.uuid AS &M.uuid, cloud_credential.invalid AS &M.invalid
 FROM    cloud_credential
-        JOIN cloud ON cloud_credential.cloud_uuid = cloud.uuid
-WHERE   cloud.name = $M.cloud_name AND cloud_credential.name = $M.credential_name AND cloud_credential.owner_uuid = $M.owner
+INNER JOIN cloud
+ON cloud_credential.cloud_uuid = cloud.uuid
+INNER JOIN user
+ON cloud_credential.owner_uuid = user.uuid
+WHERE   cloud.name = $M.cloud_name
+AND cloud_credential.name = $M.credential_name
+AND user.name = $M.owner
+AND user.removed = false
 `
 	stmt, err := sqlair.Prepare(q, sqlair.M{})
 	if err != nil {
```

Comment:
`// TODO(wallyworld) - implement owner as a FK when users are modelled.` probably should be removed right?